great_expectations/expectations/metrics/column_aggregate_metrics/column_distinct_values.py

```diff
@@ -42,49 +54,67 @@ def _spark(
         metric_value_kwargs: Dict,
         metrics: Dict[str, Any],
         runtime_configuration: Dict,
-    ):
-        observed_value_counts = metrics["column.value_counts"]
-        return set(observed_value_counts.index)
-
-    @classmethod
-    def _get_evaluation_dependencies(
-        cls,
-        metric: MetricConfiguration,
-        configuration: Optional[ExpectationConfiguration] = None,
-        execution_engine: Optional[ExecutionEngine] = None,
-        runtime_configuration: Optional[Dict] = None,
-    ):
-        """Returns a dictionary of given metric names and their corresponding configuration,
-        specifying the metric types and their respective domains"""
-        dependencies: dict = super()._get_evaluation_dependencies(
-            metric=metric,
-            configuration=configuration,
-            execution_engine=execution_engine,
-            runtime_configuration=runtime_configuration,
+    ) -> Set[Any]:
+        df, _, accessor_domain_kwargs = execution_engine.get_compute_domain(
+            metric_domain_kwargs, MetricDomainTypes.COLUMN
+        )
+        column = accessor_domain_kwargs["column"]
+
+        distinct_values = (
+            df.select(column)
+            .distinct()
+            .where(F.col(column).isNotNull())
+            .rdd.flatMap(lambda x: x)
+            .collect()
         )
 
-        if isinstance(
-            execution_engine, (SqlAlchemyExecutionEngine, SparkDFExecutionEngine)
-        ):
-            dependencies["column.value_counts"] = MetricConfiguration(
-                metric_name="column.value_counts",
-                metric_domain_kwargs=metric.metric_domain_kwargs,
-                metric_value_kwargs={
-                    "sort": "value",
-                    "collate": None,
-                },
-            )
-
-        return dependencies
+        return set(distinct_values)
 
 
 class ColumnDistinctValuesCount(ColumnAggregateMetricProvider):
     metric_name = "column.distinct_values.count"
 
     @column_aggregate_value(engine=PandasExecutionEngine)
-    def _pandas(cls, column, **kwargs):
+    def _pandas(cls, column, _metrics, **kwargs):
+        if "column.distinct_values" in _metrics:
```

Comment:
Curious to know how we feel about this conditional. If we already know the distinct values, we don't need to query again. If we only want the distinct values count, we can get that in a single step instead of forcing a dependency resolution and performing in-memory operations on that.

Reply:
I am good with this especially if `len` does an attribute lookup vs a loop in python. What is the type of the `_metrics["column.distinct_values"]`?

Reply:
@NathanFarmer Let us discuss -- I think that there is a potential "no-op"/tautology involved there (easier via zoom).  Thanks!

Reply:
This is no longer applicable since I learned that `_metrics` can only be supplied via `MetricConfiguration` `metric_dependencies` and not via `get_evaluation_dependencies` which is what is used during metric resolution.